redis.go

```diff
@@ -24,13 +24,25 @@ type Client struct {
 	client *redis.Client
 }
 
+// ClusterClient is the Redis cluster client wrapper
+type ClusterClient struct {
+	client *redis.ClusterClient
+}
+
 // XClient represents the Client constructor (i.e. `new redis.Client()`) and
 // returns a new Redis client object.
 func (r *Redis) XClient(ctxPtr *context.Context, opts *redis.Options) interface{} {
 	rt := common.GetRuntime(*ctxPtr)
 	return common.Bind(rt, &Client{client: redis.NewClient(opts)}, ctxPtr)
 }
 
+// XClusterClient represents the Client constructor (i.e. `new redis.Client()`) and
+// returns a new Redis client object.
+func (r *Redis) XClusterClient(ctxPtr *context.Context, opts *redis.ClusterOptions) interface{} {
+	rt := common.GetRuntime(*ctxPtr)
+	return common.Bind(rt, &ClusterClient{client: redis.NewClusterClient(opts)}, ctxPtr)
+}
```

Comment:
I'm not a fan of using a separate constructor and struct for this. You had to duplicate the implementations of `Set` and `Get`, and since this extension only exposes the bare minimum Redis commands, it will get very cumbersome if we want to expand it.

We could embed a base struct and avoid that, but I'm wondering if we could use the same `Client` struct and internally initialize `redis.Client` or `redis.ClusterClient` based on a parameter passed to the constructor. Since the only difference seems to be `addr` vs `addrs`, could you implement this based on a check if `addrs` is passed?